README.md

```diff
@@ -15,55 +26,31 @@ In your Docker engine you'll now see:
-* The `run.sh` script is a convenience for calling the binary with sane default arguments (**NOTE:** you can pass in a comma-separated list of test names as the argument to run just those tests)
-* The controller Docker image is produced to the Docker image tag `kurtosistech/ava-e2e-tests_controller:latest` (at time of writing) every time `scripts/rebuild_controller_image.sh` is run
-* Every time you change:
-    1. The initializer you'll need to rerun `scripts/rebuild_initializer_binary.sh`
-    2. The controller you'll need to rerun `scripts/rebuild_controller_image.sh`
-    3. The commons code you'll need to rerun both
-
-### Helpful Tip
-You can use the following alias to stop & remove Docker containers of a certain type:
+Developing Locally
+------------------
+### Architecture
+This repo uses the [Kurtosis architecture](https://github.com/kurtosis-tech/kurtosis), so you should first go through the tutorial there to familiarize yourself with the core Kurtosis concepts.
 
-```
-# alias for clearing kurtosis containers 
-clear_containers() {  docker rm $(docker stop $(docker ps -a -q --filter ancestor="$1" --format="{{.ID}}")); } 
-alias cclear=clear_containers
-```
+In this implementation of Kurtosis, we have:
+* `AvaService` and `GeckoService` as our service interfaces
+* `GeckoServiceInitializerCore` and `GeckoServiceAvailabilityChecker` for including Gecko services in test networks
+    * `GeckoCertProvider` to allow controlling the cert that a Gecko service starts with, to allow for writing duplicate-node-ID tests
+* `TestGeckoNetwork` to encapsulate a Gecko test network of arbitrary size
+* Several tests
+* `AvaTestSuite` to contain all the tests Kurtosis can run
+* A `main.go` for running a controller Docker image under the `controller` package
+* A `main.go` for running the Kurtosis initializer under the `initializer` package
 
-Example for clearing the test nodes and test controllers:
-```
-export GECKO_IMAGE=gecko-684ca4e
-export CONTROLLER_IMAGE=kurtosistech/ava-e2e-tests_controller
-# run the tests
-./build/ava-e2e-tests -gecko-image-name="${GECKO_IMAGE}" -test-controller-image-name="${CONTROLLER_IMAGE}"
-# ...Ctrl-C to kill the test CLI...
-# clear the docker containers initialized by the tests
-cclear ${GECKO_IMAGE} 
-cclear ${CONTROLLER_IMAGE} 
-```
-
-## Writing Tests
-This repo uses the [Kurtosis architecture](https://github.com/kurtosis-tech/kurtosis), so you'll want to be familiar with the concepts there. In this implementation:
-* The `AvaTestSuite` struct defines the tests that will be run
-* The `AvaTestSuite` struct gets registered with the initializer and the controller in `initializer/main.go` and `controller/main.go` respectively
-* The tests, the networks the tests will run against, and the services the networks are composed of live in the `commons` package
+Additionally, this repo also contains a Gecko client (which should probably be moved to the Gecko repo).
```

Comment:
worth describing this as an golang RPC to differentiate from "Gecko client" as in gecko itself